errcheck/config/internal/legacy/config.go

```diff
@@ -37,15 +40,40 @@ func UpgradeConfig(cfgBytes []byte) ([]byte, error) {
 	}
 
 	var v0Cfg v0.Config
-	if legacyCfg.Args[0] != "-ignore" {
-		return nil, errors.Errorf(`errcheck-asset only supports legacy configuration if the first element in "args" is "-ignore"`)
-	}
-	for _, currIgnore := range legacyCfg.Args[1:] {
-		v0Cfg.Ignore = append(v0Cfg.Ignore, currIgnore)
+
+	args := legacyCfg.Args
+	for i := 0; i < len(args); i++ {
+		switch args[i] {
+		case "-ignore":
+			i++
+			if i >= len(args) {
+				break
+			}
+
+			v0Cfg.Ignore = append(v0Cfg.Ignore, args[i])
```

Comment:
This only adds the first argument -- shouldn't the rest of the arguments be added as well?

Reply:
As best as I can tell, `errcheck` reads a single value for the `-ignore` flag that is a comma-separated list of ignore patterns. Any functional configuration that specified the `-ignore` flag should only have a single value.